pkg/skaffold/schema/validation/validation.go

```diff
@@ -570,44 +651,61 @@ func validateLogPrefix(lc latestV1.LogsConfig) []error {
 // validateCustomTest
 // - makes sure that command is not empty
 // - makes sure that dependencies.ignore is only used in conjunction with dependencies.paths
-func validateCustomTest(tcs []*latestV1.TestCase) (errs []error) {
-	for _, tc := range tcs {
-		for _, ct := range tc.CustomTests {
+func validateCustomTest(cfg *parser.SkaffoldConfigEntry, tcs []*latestV1.TestCase) (cfgErrs []ErrorWithLocation) {
+	for i, tc := range tcs {
+		for j, ct := range tc.CustomTests {
 			if ct.Command == "" {
-				errs = append(errs, fmt.Errorf("custom test command must not be empty;"))
+				cfgErrs = append(cfgErrs, ErrorWithLocation{
+					Error:    fmt.Errorf("custom test command must not be empty;"),
+					Location: cfg.YAMLInfos.Locate(&cfg.Test[i].CustomTests[j]),
+				})
 				return
 			}
 
 			if ct.Dependencies == nil {
 				continue
 			}
 			if ct.Dependencies.Command != "" && ct.Dependencies.Paths != nil {
-				errs = append(errs, fmt.Errorf("dependencies can use either command or paths, but not both"))
+				cfgErrs = append(cfgErrs, ErrorWithLocation{
+					Error:    fmt.Errorf("dependencies can use either command or paths, but not both"),
+					Location: cfg.YAMLInfos.Locate(&cfg.Test[i].CustomTests[j]),
+				})
 			}
 			if ct.Dependencies.Paths == nil && ct.Dependencies.Ignore != nil {
-				errs = append(errs, fmt.Errorf("customTest has invalid dependencies; dependencies.ignore can only be used in conjunction with dependencies.paths"))
+				cfgErrs = append(cfgErrs, ErrorWithLocation{
+					Error:    fmt.Errorf("customTest has invalid dependencies; dependencies.ignore can only be used in conjunction with dependencies.paths"),
+					Location: cfg.YAMLInfos.Locate(&cfg.Test[i].CustomTests[j]),
+				})
 			}
 		}
 	}
 	return
 }
 
-func wrapWithContext(config *parser.SkaffoldConfigEntry, errs ...error) []error {
-	var id string
+func wrapWithContext(config *parser.SkaffoldConfigEntry, errs ...ErrorWithLocation) []ErrorWithLocation {
+	module := ""
 	if config.Metadata.Name != "" {
-		id = fmt.Sprintf("in module %q", config.Metadata.Name)
-	} else {
-		id = fmt.Sprintf("in unnamed config at index %d", config.SourceIndex)
+		module = fmt.Sprintf(" in module %q", config.Metadata.Name)
 	}
+	// else {
+	// 	// TODO(aaron-prindle) what is config.SourceIndex? Seems it is 0 in my testing usually...
+	// 	id = fmt.Sprintf("in unnamed config at index %d", config.SourceIndex)
+	// }
```

Comment:
should this be removed?

Reply:
Yes, changed this to respect how it was before and comments are removed